gp/primitives.py

```diff
@@ -26,33 +23,24 @@ def __init__(self, primitives: List[Type[Node]]) -> None:
             if not issubclass(primitives[i], Node):
                 raise TypeError(f"expected subclass of Node but received {primitives[i].__name__}")
 
-        self._primitives = {}
-        for i in range(len(primitives)):
-            self._primitives[i] = primitives[i]
-
-        # hide primitives dict behind MappingProxyType to make sure it
-        # is not changed after construction
-        # unfortunately not supported by pickle, necessary for
-        # multiprocessing; another way to implement this?
-        # self._primitives = types.MappingProxyType(self._primitives)
+        self._primitives = list(primitives)
+        self._max_arity: int = self._determine_max_arity()
 
-        self._determine_max_arity()
+    def __iter__(self) -> Iterator[Type[Node]]:
+        return iter(p for p in self._primitives)
```

Comment:
why not `iter(self._primitives)`?

Reply:
good point. but even better: we don't need this function any more :)

Reply:
Hmm, I think we do need it so that `list(Primitives)` works. That's also why `mypy` complains in CI.

Reply:
it's not needed for `list()` (`__getitem__`) is enough, but i will look into the mypy issue